_posts/2022-08-03-#25527.md

```diff
@@ -7,19 +7,108 @@ authors: [dongcarl]
+
+    (Hint b) What needs to be called/initialized for the test to use the script cache?
+
+    (Hint c) The checkinputs_test test case uses `Dersig100Setup`. How can you check if this has a script cache setup?
+
+1. Describe the uint32 overflow in [this
+   commit](https://github.com/bitcoin/bitcoin/pull/25527/commits/3c5555de81ab7f51c655dadffc5e939c4515f65d). Under what conditions would it get hit?
+
+1. [This commit](https://github.com/bitcoin-core-review-club/bitcoin/commit/f66cd5b3aa25f27638b05fbffde7470dd844951b) returns approximate total size from
+   [`InitSignatureCache()`](https://github.com/bitcoin-core-review-club/bitcoin/commit/f66cd5b3aa25f27638b05fbffde7470dd844951b#diff-0618dae2990d096e96e0283f7ab8cee069469f1ce603b58c0bb289e154f3aa17R36)
+and
+[`InitScriptExecutionCache`](https://github.com/bitcoin-core-review-club/bitcoin/commit/f66cd5b3aa25f27638b05fbffde7470dd844951b#diff-d3c243938494b10666b44404a27af7d84b44a72b85a27431e0c89e181462ca6eR327).
+It also adds the [`[[nodiscard]]`](https://en.cppreference.com/w/cpp/language/attributes/nodiscard)
+attribute to both functions. What does `[[nodiscard]]` do, and why
+or why not is it appropriate to add here?
+
+1. [This
+   commit](https://github.com/bitcoin-core-review-club/bitcoin/commit/d29f2fd9b49ac00a3721af7260dbbf59e9e8387c)
+changes the type of `signature_cache_bytes` and `script_execution_cache_bytes` from `int64_t` to
+`size_t`. What is the difference between `int64_t`, `uint64_t`, and `size_t`, and why should
+a `size_t` hold these values?
+
+1. What is the maximum value for `ValidationCacheSizes::signature_cache_bytes` after [this
+   diff](https://github.com/bitcoin-core-review-club/bitcoin/commit/d29f2fd9b49ac00a3721af7260dbbf59e9e8387c#diff-f36fa9dcd326237b98e6d365b03eea9af04bccd5a555fdf91f402cafc7d20cdcR26-R29)?
+Is this a behavior change? Why or why not is the code correct?
+
+1. The
+   [commit](https://github.com/bitcoin-core-review-club/bitcoin/commit/e4d5bab88e9acad53cb8d636d836d2cdf9bb49b8)
+introducing src/node/validation\_cache\_args
+[adds](https://github.com/bitcoin-core-review-club/bitcoin/commit/e4d5bab88e9acad53cb8d636d836d2cdf9bb49b8#diff-5aa4eb91717e30e73956da744da7d89de289d4bcca940b73a6285ea547275a94R47)
+it to the include-what-you-use check in the tidy CI job. What does this check do? Were you able to
+run it yourself?
+
+1. Quiz: A mainnet node receives a new block at height 710,000 containing a few
+   transactions (with taproot activation height at 709,632). Which of the following transactions
+could hit its script cache, assuming nothing has been evicted?
 
+```
+(A) the coinbase transaction
+(B) a transaction the node has never seen before
+(C) a transaction with the same txid but different witness as a tx in its mempool
+(D) a transaction that was in its mempool but replaced by another
+(E) a non-taproot transaction accepted to the mempool at height 709,000
```

Comment:
nit: I think technically there's no such thing as (non-) taproot transactions? Sacrifices readability though, so maybe keep it?
```suggestion
(E) a transaction without taproot inputs and outputs accepted to the mempool at height 709,000
```